HACKING.md

```diff
@@ -0,0 +1,388 @@
+
+    python -m pip install sphinx
+
+We'll use this later.
+
+### `pre-commit` Setup
+
+Here at Salt we use [pre-commit](https://pre-commit.com/) to make it easier for
+contributors to get quick feedback. It configures some Git pre-commit hooks to
+run `black` for formatting, `isort` for keeping our imports sorted, and
+`pylint` to catch issues like unused imports, among others. You can easily
+install it in your virtualenv with:
+
+    python -m pip install pre-commit
+    pre-commit init
+
+Now before each commit, it will ensure that your code at least *looks* right
+before you open a pull request. And with that step, it's time to start hacking
+on Salt!
+
+
+## Selecting an Issue
+
+If you don't already have an issue in mind, you can search for [help
+wanted](https://github.com/saltstack/salt/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22)
+issues. If you also search for [good first
+issue](https://github.com/saltstack/salt/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22+label%3A%22good+first+issue%22)
+then you should be able to find some issues that are good for getting started
+contributing to Salt. [Documentation
+issues](https://github.com/saltstack/salt/issues?q=is%3Aissue+is%3Aopen+label%3Adocumentation+)
+are also good starter issues. When you find an issue that catches your eye (or
+one of your own), it's a good idea to comment on the issue and mention that
+you're working on it. Good communication is key to collaboration - so if you
+don't have time to complete work on the issue, just leaving some information
+about when you expect to pick things up again is a great idea!
+
+## Hacking Away
+
+### Salt, Tests, Documentation, and You
+
+To merge code contributions, Salt requires:
+
+- documentation
+- meaningful passing tests
+- correct code
+
+Documentation fixes just require correct documentation.
+
+#### What If I Don't Write Tests or Docs?
+
+If you aren't into writing documentation or tests, we still welcome your
+contributions! But your PR will be labeled `Needs Testcase` and `Help Wanted`
+until someone can get to write the tests/documentation. Of course, if you have
+a desire but just lack the skill we are more than happy to collaborate and help
+out! There's the [documentation working
+group](https://github.com/saltstack/docs-hub) and the [testing working
+group](https://github.com/saltstack/community/tree/master/working_groups/wg-Testing).
+We also regularly stream our test clinic [live on
+Twitch](https://www.twitch.tv/saltstackinc) every Tuesday afternoon and
+Thursday morning, Central Time. If you'd like specific help with tests, bring
+them to the clinic.  If no community members need help, you can also just watch
+tests written in real time.
+
+### Documentation
+
+Salt uses both docstrings, as well as normal reStructuredText files in the
+`salt/doc` folder for documentation. To build the docs and view them in your
+browser, simply run:
+
+    cd doc; make html; cd _build/html; python -m http.server
```

Comment:
Again, the nox docs target

Reply:
Or use `nox -e 'docs-html(compress=False, clean=True)'` (`nox` commands can be viewed with `nox -l`)

Reply:
☝️

Reply:
also added